cuckoo/root_node.py

```diff
@@ -78,123 +92,96 @@ def _get_response(
-            return self._session_async
-        else:
-            if self._session is None:
-                self._close_session = True
-                self._session = Client(timeout=None)
-            return self._session
-
-    def _process_response(
-        self,
-        response: Response,
-        query: str,
-        variables: dict,
-    ):
-        response.raise_for_status()
-        response_json: dict[str, Any] = response.json()
+    def _process_response(self):
+        response_json: dict[str, Any] = self._response.json()
+        if self._logger:
+            response_text = to_truncated_str(self._response.text)
+            request_text = to_truncated_str(self._response.request.content.decode())
 
         if ("error" in response_json) or ("errors" in response_json):
             errors: list[dict[str, str]] = (
                 [response_json.get("error")] if "error" in response_json else []
             ) + response_json.get("errors", [])
             if self._logger:
                 self._logger.error(
-                    f"Query failed. query={self._compact(query)}, "
-                    f"variables={to_truncated_str(variables)}, response={str(errors)}."
+                    f"Query failed.  Request={request_text}, "
+                    f"response={response_text}, errors={str(errors)}."
                 )
             raise HasuraServerError(errors)
-
         elif "data" in response_json:
             if self._logger:
                 self._logger.debug(
-                    f"Query successful. query={self._compact(query)}, "
-                    f"variables={to_truncated_str(variables)}, "
-                    f"response={to_truncated_str(response_json['data'])}."
+                    f"Query successful. Request={request_text}, "
+                    f"response={response_text}."
                 )
-            return response_json["data"]
-
+            self._response_data = response_json["data"]
         else:
             raise NotImplementedError(
-                "response dict did not contain any errors nor data."
+                "Response did not contain any errors nor data. "
+                f"Response={self._response.text}."
             )
 
-    def _execute(self: RootNode):
-        query = str(self)
+    def _build_request(self):
+        query = to_compact_str(str(self))
         variables = self._get_all_variables()
-        json = {"query": self._compact(query)}
+        json = {"query": query}
         if variables:
             json["variables"] = RootNode._jsonify_variables(variables)
 
         if self._logger:
             self._logger.debug(
-                f"Executing query. query={self._compact(query)}, "
-                f"variables={to_truncated_str(variables)}."
+                f"Request created. {query=}, variables={to_truncated_str(variables)}."
             )
 
-        session = self._get_or_create_session()
+        return self._session.build_request(
+            method="POST",
```

Comment:
cleanup refactor:
note that with `httpx`, you can either do
```py
session.post(**args)
# OR
request=session.build_request(method="POST", **args)
session.send(request)
```

since we want always the same request, but send it in different ways (async, stream), I use the second way now